File: src/DistinctNumbers.java
//Link: https://codeforces.com/group/MWSDmqGsZm/contest/223205/problem/M

import java.util.ArrayList;
import java.util.HashSet;
import java.util.Scanner;

public class DistinctNumbers
{
    public  static void main(String[]args)
    {
        Scanner cs=new Scanner(System.in);
        int n=cs.nextInt();
        ArrayList<Integer>v=new ArrayList<>();
        for(int i=0;i<n;i++)
        {
            v.add(cs.nextInt());
        }
        distNumber(v);
    }

    private static void distNumber(ArrayList<Integer>v)
    {
        HashSet<Integer> s=new HashSet<>(v);
        System.out.print(s.size());
    }
}

File: function/Distinct Numbers.cpp
//Link: https://codeforces.com/group/MWSDmqGsZm/contest/223205/problem/M

#include <iostream>
#include <algorithm>
#include<cmath>
#include <vector>
#include <iomanip>
#include <set>
using namespace std;

void distNumber(vector<int>&v)
{
	set<int>s(v.begin(), v.end());
	cout << s.size() << endl;
}

int main()
{
	int n;
	cin >> n;
	vector<int>v(n);
	for (int i = 0; i < n; i++)
	{
		cin >> v[i];
	}
	distNumber(v);
	

}